synapse/notifier.py

```diff
@@ -550,3 +550,9 @@ def wait_for_replication(self, callback, timeout):
                 break
 
         defer.returnValue(result)
+
+    def wait_once_for_replication(self):
+        """Returns a deferred which resolves when there is new data for
+        replication to handle.
+        """
+        return self.replication_deferred.observe()
```

Comment:
It may be nicer to let the replication resource register a callback rather than bouncing through a deferred?